cmd/quadlet/main.go

```diff
@@ -491,6 +491,14 @@ func enableServiceFile(outputPath string, service *parser.UnitFile) {
 				symlinks = append(symlinks, fmt.Sprintf("%s.requires/%s", requiredByUnit, serviceFilename))
 			}
 		}
+
+		upheldBy := service.LookupAllStrv(quadlet.InstallGroup, "UpheldBy")
+		for _, upheldByUnit := range upheldBy {
+			// Only allow filenames, not paths
+			if !strings.Contains(upheldByUnit, "/") {
+				symlinks = append(symlinks, fmt.Sprintf("%s.upholds/%s", upheldByUnit, serviceFilename))
+			}
+		}
```

Comment:
The code between WantedBy, RequiredBy and UpheldBy is the same. I think it would be best to split the common part out in a function.